src/initialization/MOM_shared_initialization.F90

```diff
@@ -413,17 +413,27 @@ subroutine limit_topography(D, G, param_file, max_depth, US)
                  "The depth below which to mask the ocean as land.", &
                  units="m", default=-9999.0, scale=m_to_Z, do_not_log=.true.)
 
-! Make sure that min_depth < D(x,y) < max_depth
-  if (mask_depth < -9990.*m_to_Z) then
-    do j=G%jsd,G%jed ; do i=G%isd,G%ied
-      D(i,j) = min( max( D(i,j), 0.5*min_depth ), max_depth )
-    enddo ; enddo
+  if (mask_depth > min_depth) then
+    mask_depth = -9999.0*m_to_Z
+    call MOM_error(WARNING, "MOM_shared_initialization: limit_topography "//&
+                  'MASKING_DEPTH is larger than MINIMUM_DEPTH and therefore ignored.')
+  endif
+
+  ! Make sure that min_depth < D(x,y) < max_depth for ocean points
+  if (abs(mask_depth - (-9999.*m_to_Z)) < 1.0e-10) then
```

Comment:
The expression on line 423 has problems with the dimensional rescaling.  It should be `if (abs(mask_depth - (-9999.*m_to_Z)) < 1.0e-10*m_to_Z) then`.